apps/console/src/features/roles/components/edit-role/edit-role-users.tsx

```diff
@@ -124,36 +117,57 @@ export const RoleUsersList: FunctionComponent<RoleUsersPropsInterface> = (
         !!selectedUserStoreDomainName || !!userSearchValue
     );
 
+    const isUserBelongToSelectedUserStore = (user: UserBasicInterface, userStoreName: string) => {
+        const userNameChunks: string[] = user.userName.split("/");
+
+        return (userNameChunks.length === 1 && userStoreName === "Primary")
+        || (userNameChunks.length === 2 && userNameChunks[0] === userStoreName.toUpperCase());
+    };
+
     useEffect(() => {
-        const alreadyAssignedUsers: UserBasicInterface[] = role?.users?.map( (user: RolesMemberInterface) => {
-            return {
-                id: user.value,
-                userName: user.display
-            };
-        });
-
-        if (alreadyAssignedUsers?.length > 0) {
-            setSelectedUsers([
-                ...alreadyAssignedUsers
-            ]);
+        if (!role?.users?.length) {
+            return;
         }
-    },[ role, selectedUserStoreDomainName ]);
 
-    useEffect(() => {
-        if (selectedUsers?.length > 0) {
-            setIsPlaceholderVisible(false);
+        setIsPlaceholderVisible(false);
+
+        const alreadyAssignedUsersToSelectedUserStore: UserBasicInterface[] = role?.users?.map(
+            (user: RolesMemberInterface) => {
+                return {
+                    id: user.value,
+                    userName: user.display
+                };
+            }).filter((user: UserBasicInterface) => {
+
+            const belongToUserStore: boolean = isUserBelongToSelectedUserStore(user, selectedUserStoreDomainName);
+
+            if (belongToUserStore) {
+                return user;
+            }
+        }).filter(Boolean) ?? [];
+
+        const tempSelectedAllUsers: Record<string, UserBasicInterface[]> = selectedAllUsers;
+
+        if (tempSelectedAllUsers[selectedUserStoreDomainName] === undefined) {
+            setSelectedUsersFromUserStore(alreadyAssignedUsersToSelectedUserStore);
+        } else {
+            setSelectedUsersFromUserStore(tempSelectedAllUsers[selectedUserStoreDomainName]);
         }
```

Comment:
Suggestion to simplify the logic. Test the flow accordingly.

```
const alreadyAssignedUsersToSelectedUserStore: UserBasicInterface[] = role?.users?.map(
    (user: RolesMemberInterface) => ({
        id: user.value,
        userName: user.display
    })
).filter((user: UserBasicInterface) => 
    isUserBelongToSelectedUserStore(user, selectedUserStoreDomainName)
) ?? [];

const usersFromSelectedStore = selectedAllUsers[selectedUserStoreDomainName] || alreadyAssignedUsersToSelectedUserStore;

setSelectedUsersFromUserStore(usersFromSelectedStore);
```

Reply:
Thanks. Fixed